crates/autopilot/src/run_loop.rs

```diff
@@ -176,11 +184,18 @@ impl RunLoop {
         Some(domain::AuctionWithId { id, auction })
     }
 
-    async fn single_run(&self, auction_id: domain::auction::Id, auction: &domain::Auction) {
+    async fn single_run(
+        self: Arc<Self>,
+        auction_id: domain::auction::Id,
+        auction: &domain::Auction,
+    ) {
         let single_run_start = Instant::now();
         tracing::info!(?auction_id, "solving");
 
-        let solutions = self.competition(auction_id, auction).await;
+        let auction = self.remove_in_flight_orders(auction.clone()).await;
```

Comment:
Looks like we can avoid the clone by passing an owner `Auction` into `single_run()`.